kickstarter/android-oss
# MBL-831: SearchActivity-ViewModel migrated to RXJava2

# 📲 What

- `SearchActivity` + `SearchViewModel` migrated to RXJava2 + adopted jetpack ViewModel
- Got rid of the `rxbinding.recyclerView` dependency usage for this specific activity/viewModel convo by creating a custom subscriber for recyclerView scroll events for pagination.

# 🤔 Why

- Better handling of lifecycle events
- Dispose of subscriptions to avoid memory leaks
- Use compose if we want to

# 🛠 How

If you need more details about the how, take a look to the details [here](https://kickstarter.atlassian.net/wiki/spaces/NT/pages/2350317579/ViewModel+Tech+Debt+Epic#Technical-details)

# 👀 See


https://github.com/kickstarter/android-oss/assets/4083656/703f9499-ad2e-49c3-a263-4b37eab3b0b7



|  |  |

# 📋 QA

- Go to search activity, without typing any search term you should see popular projects loading
- Scroll until you get more results to double check the infinite scroll(pagination) works properly for the popular projects
- type something to search, scroll until you get more results for a search term.
- You can compare any of the previous described behaviour for this build with the production app, nothing should have changed.

# Story 📖

[MBL-831](https://kickstarter.atlassian.net/browse/MBL-831)


[MBL-831]: https://kickstarter.atlassian.net/browse/MBL-831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
